feat(workspace-tree): Sync open editor with workspace tree view

cbandera opened the pull request:
This PR implements a synchronization between the file opened in the editor and the workspace tree view of the bazel extension.

Note on implementation:
The implementation of reveal() in the tree view required proper parent-child relationships and package path resolution. The changes ensure that:

1. The tree view can properly navigate and reveal items by maintaining correct parent references
2. Package paths are consistently resolved through the parent chain
3. The selected item in the tree view is synchronized with the currently open editor

A cache of BazelPackageTreeItems was added to improve performance by:

- Reducing redundant package path calculations
- Avoiding bazel queries performed by existing getChildren operator
- Minimizing repeated string operations when resolving package paths

This PR is part of #353

cbandera commented:
@cameron-martin , @vogelsgesang I assumed the tests might be failing because the github runner env doesn't have `bazel` installed, but adding it to the workflow file didn't solve the problem. The tests run through locally (even though I got the same error once). So they might also be flaky - I have the feeling it could have something to do with timing. Maybe we need to wait longer inside the testcase, but I am unsure how and where to do it. 
Any hints on how to debug this are welcome.

cbandera commented:
@cameron-martin , @vogelsgesang I have deactivated the one problematic test case for now, so that this PR becomes green anyway. 
As stated before, the feature works when manually starting the extension and performing this test case, I just can't get the automated test to work, probably due to timing/concurrency issues.
If you have the time to analyze or even fix the testcase, feel free to do so. Otherwise I am also happy for any advice you can give.

cameron-martin commented:
Looks like the tests are failing even though you disabled the problematic test. Could you take a look at this?

cbandera commented:
Ok found it and fixed it. The problem was that due to the way the test was setup, there were multiple instances of the treeview provider which messed things up.
I now changed it to use the one single instance of the running extension for testing. Which fixes all tests.

cameron-martin commented on src/workspace-tree/bazel_workspace_tree_provider.ts:
```diff
@@ -55,17 +63,36 @@ export class BazelWorkspaceTreeProvider
       false,
     );
+
+    // Initialize disposables array if it's not already initialized
```
What's the point of initialising it like this, rather than like how it used to be?

cameron-martin commented on src/workspace-tree/bazel_workspace_tree_provider.ts:
```diff
@@ -158,8 +185,103 @@ export class BazelWorkspaceTreeProvider
+    }
+
+    const relativeFilePath = path.relative(
+      workspaceFolderVSCode.uri.fsPath,
```
That's a good question. It'd be nice if we didn't have to assume a 1:1 mapping, but yeah I imagine we're quite inconsistent/buggy in this respect. For example sometimes we run bazel commands in the workspace that the current file resides in, instead of the vscode workspace root, causing issues like #383.

Happy to defer this to later though. From prior discussions I've had doing workspace discovery "correctly" is hard.
